firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -360,9 +360,20 @@ class BrowserViewController: UIViewController,
         let isActionNeeded = RustFirefoxAccounts.shared.isActionNeeded
         let showWarningBadge = consume isActionNeeded
 
-        if isToolbarRefactorEnabled && showWarningBadge {
-            let action = ToolbarMiddlewareAction(windowUUID: windowUUID, actionType: ToolbarActionType.showMenuWarningBadge)
-            store.dispatch(action)
+        if isToolbarRefactorEnabled {
+            if showWarningBadge {
+                let action = ToolbarMiddlewareAction(
```

Comment:
Why do we dispatch a `ToolbarMiddlewareAction` here? Wouldn't it be easier to dispatch the `ToolbarAction` with or without the badge image here right away?